services/src/pro/datasets/add_from_directory.rs

```diff
@@ -0,0 +1,88 @@
+use std::{
+    fs::{self, DirEntry, File},
+    io::BufReader,
+    path::PathBuf,
+};
+
+use crate::error::Result;
+use crate::{
+    datasets::storage::DatasetDb,
+    pro::datasets::{DatasetPermission, Permission, Role},
+};
+use crate::{
+    datasets::storage::DatasetDefinition, pro::users::UserSession, util::user_input::UserInput,
+};
+
+use log::warn;
+
+use super::storage::UpdateDatasetPermissions;
+
+pub async fn add_datasets_from_directory<D: DatasetDb<UserSession> + UpdateDatasetPermissions>(
+    db: &mut D,
+    file_path: PathBuf,
+) {
+    async fn add_dataset_definition_from_dir_entry<
+        D: DatasetDb<UserSession> + UpdateDatasetPermissions,
+    >(
+        db: &mut D,
+        entry: &DirEntry,
+        system_session: &UserSession,
+    ) -> Result<()> {
+        let def: DatasetDefinition =
+            serde_json::from_reader(BufReader::new(File::open(entry.path())?))?;
+
+        let dataset_id = db
+            .add_dataset(
+                system_session,
+                def.properties.validated()?,
+                db.wrap_meta_data(def.meta_data),
+            )
+            .await?;
+
+        db.add_dataset_permission(
+            system_session,
+            DatasetPermission {
+                role: Role::user_role_id(),
+                dataset: dataset_id.clone(),
+                permission: Permission::Read,
+            },
+        )
+        .await?;
+
+        db.add_dataset_permission(
+            system_session,
+            DatasetPermission {
+                role: Role::anonymous_role_id(),
+                dataset: dataset_id,
+                permission: Permission::Read,
+            },
+        )
+        .await?;
```

Comment:
So there is a generic user role id? This sounds strange. Couldn't we make a public role that only requires a single assignment? Maybe you have to explain your rationale behind this. Or maybe make some doc-comments.